### ISSUE TITLE
Obsolete "torch.autograd.Variable" in deeprank.learn.modelGenerator

### ISSUE THREAD

**Issue by DarioMarzella** [DeepRank/deeprank#175]
**Is your feature request related to a problem? Please describe.**
In `deeprank.learn.modelGenerator`  (and, sequentially, in  all generated models like `deeprank.lean.model3d ` and ` deeprank.lean.model2d `) in the `_get_conv_output` function, the method `torch.autograd.Variable` is used to keep track of the gradients. 
This method is outdated.

**Describe alternatives you've considered**
torch.autograd.Variable could be replaced by simply setting 'requires_grad = True' for the output tensor
